simulator/src/main/java/com/hedera/block/simulator/grpc/PublishStreamObserver.java

```diff
@@ -27,21 +29,28 @@
 public class PublishStreamObserver implements StreamObserver<PublishStreamResponse> {
 
     private final Logger logger = System.getLogger(getClass().getName());
-
-    /** Creates a new PublishStreamObserver instance. */
-    public PublishStreamObserver() {}
+    private final AtomicBoolean allowNext;
+
+    /** Creates a new PublishStreamObserver instance.
+     *
+     * @param allowNext is responsible for signaling, whether streaming should continue
+     */
+    public PublishStreamObserver(final AtomicBoolean allowNext) {
+        this.allowNext = allowNext;
+    }
 
     /** what will the stream observer do with the response from the server */
     @Override
     public void onNext(PublishStreamResponse publishStreamResponse) {
         logger.log(Logger.Level.INFO, "Received Response: " + publishStreamResponse.toString());
     }
 
-    /** what will the stream observer do when an error occurs */
+    /** Responsible for stream observer behaviour, in case of error. For now, we will stop the stream for every error. In the future we'd want to have a retry mechanism depending on the error. */
     @Override
     public void onError(Throwable throwable) {
-        logger.log(Logger.Level.ERROR, "Error: " + throwable.toString());
-        // @todo(286) - handle the error
+        Status status = Status.fromThrowable(throwable);
+        logger.log(Logger.Level.ERROR, "Error: " + throwable.toString() + " with status code: " + status.getCode());
+        allowNext.set(false);
```

Comment:
Maybe put `allowNext.set(false)` as the first call in this method. We want to shut it down as quickly as possible.

Reply:
Done ✅